src/nunavut/lang/cpp/support/VariableLengthArray.hpp

```diff
@@ -0,0 +1,464 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * Copyright (C) 2014 Pavel Kirienko <pavel.kirienko@gmail.com>
+ * Copyright (C) 2021  UAVCAN Development Team  <uavcan.org>
+ * This software is distributed under the terms of the MIT License.
+ */
+
+#ifndef NUNAVUT_SUPPORT_VARIABLE_LENGTH_ARRAY_HPP_INCLUDED
+#define NUNAVUT_SUPPORT_VARIABLE_LENGTH_ARRAY_HPP_INCLUDED
+
+#include <cstddef>
+#include <cstring>
+#include <type_traits>
+#include <memory>
+#include <utility>
+
+namespace nunavut
+{
+namespace support
+{
+/**
+ * Minimal, generic container for storing UAVCAN dynamic arrays. One property that is unique
+ * for dynamic arrays is that they have a maximum bound which this implementation enforces.
```

Comment:
```suggestion
 * Minimal, generic container for storing UAVCAN variable-length arrays. One property that is unique
 * for variable-length arrays is that they have a maximum bound which this implementation enforces.
```

"Dynamic" may be misleading because it might be understood as a storage qualifier rather than size qualifier. The Specification never says "dynamic" for this reason, it says "variable-length".